### ISSUE TITLE
Data Validation in XML reader

### ISSUE THREAD

**Issue by SlowFox71** [PHPOffice/PhpSpreadsheet#3592]
This is:

```
- [ ] a bug report
- [x] a feature request
- [x] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
Respect datavalidation in the XML reader

### What is the current behavior?
Validation is not yet parsed.

### What features do you think are causing the issue

- [X] Reader

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?
Only XML format is affected

### Which versions of PhpSpreadsheet and PHP are affected?
All versions are affected

I attach a simple XML file which should cover the most relevant cases - if you need anything more just ask.
As I am still stuck with an ancient version I implemented this myself (quite rude, but it still might be used as a starting point - actually it seems easier than I expected. $children_x are the Excel-namespaced children of the worksheet).

Care must be taken with the "hide"-Options: XML uses "show" as a default and can only hide; in PhpSpreadsheet the defaults are set the other way round.

One big issue seems to be performance if ranges are used. Column A in the sample file uses a simple, single validation rule, column B uses one with exception of B1 - the XML reflects how Excel thinks of this. I have not found a way to replicate this into PhpSpreadsheet, but using Coordinate::extractAllCellReferencesInRange() to split ranges into single cells is really expensive.
[validation.txt](https://github.com/PHPOffice/PhpSpreadsheet/files/11611320/validation.txt)
[validate.txt](https://github.com/PHPOffice/PhpSpreadsheet/files/11611323/validate.txt)

**Comment by oleibman** [comment]
In order to avoid the performance problem that you mention with ranges, the Xlsx Reader calls `$sheet->shrinkRangeToFit($cell)`; this causes initialization of cells only up to the maximum row and column used on a sheet, not the maximum row and column in the range. That could easily be done with your code as well. It is certainly an imperfect solution. I don't know if it can be improved upon; I can imagine many complications.

**Comment by oleibman** [comment]
At any rate, you have created an unusually good example. One of your data validations exposes a nasty problem. Your "any of the texts in E1:E5", which Xlsx would store as `<formula1>E1:E5</formula1>`, Xml stores as `<Value>R[-1]C[1]:R[3]C[1]</Value>`. If we load the Xml file, and save it as Xlsx, passing the Xml Value as an Xlsx formula unchanged, Excel will complain that the Xlsx file is corrupt. Identifying when the Value needs to be changed and performing the change are probably activities that occur elsewhere in the codebase, but it will take a while to pin down, and to test adequately.

**Comment by SlowFox71** [comment]
Thanks for the hint to shrinkRangeToFit. This does the trick for my purposes, though in general it could be a problem, but for all formats.

To explore this bug I added a simple formula to my test sheet:
[validation.txt](https://github.com/PHPOffice/PhpSpreadsheet/files/11635466/validation.txt)
If I open it with Excel and save as Xlsx, both formula and validation are converted to A1 notation.

If I open it with PhpSpreadsheet and save as Xlsx the following file is produced:
[validation_result.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/11635480/validation_result.xlsx)

PhpSpreadsheet tries to convert the formula but leaves the validation rule unchanged - is there any syntactical difference or could the same code be used? But in either way the code seems to be flawed (see the example) as PhpSpreadsheet ruins the formula. If helpful and if I did not make any mistake during conversion, I could open a separate bug report for this.
